src/goose/specs.clj

```diff
@@ -0,0 +1,117 @@
+(ns goose.specs
+  (:require
+    [goose.client :as c]
+    [goose.defaults :as d]
+    [goose.utils :as u]
+    [goose.worker :as w]
+
+    [clojure.spec.alpha :as s]
+    [clojure.spec.test.alpha :as st]
+    [clojure.string :as string]
+    [taoensso.nippy :as nippy]
+    [taoensso.carmine.connections :refer [IConnectionPool]]))
+
+; ========== Qualified Function Symbols ==============
+(defn- resolvable-fn? [func] (fn? @(resolve func)))
+(s/def ::fn-sym (s/and qualified-symbol? resolve resolvable-fn?))
+
+; ========== Redis ==============
+; Valid Redis URL patterns:
+; #1: redis://username:password@hostname:0-65353
+; #2: redis://hostname:0-65353
+(s/def :redis/url #(re-matches #"redis://.+:[0-9]{1,5}" %))
+(s/def :redis/pool-opts
+  (s/or :none #(= :none %)
+        :map #(map? %)
+        :iconn-pool #(satisfies? IConnectionPool %)))
+
+; ============== Brokers ==============
+(s/def :broker/redis
+  (s/keys :req-un [:redis/url]
+          :opt-un [:redis/pool-opts]))
+(s/def ::broker-opts
+  (s/or :redis (s/keys :req-un [:broker/redis])))
```

Comment:
Why the `s/or` here? This could just be `s/keys`, right?

Reply:
Broker opts can be 
`{:redis {:url "redis://..."}}`
or 
`{:rabbitmq {:url "..."}}`

Reply:
Changed the spec to do a check so that :redis or :rabbitmq is mutually exclusive.